### PR TITLE
Panel optimize

### PR DESCRIPTION
**Pull request by set-soft**
When processing a panel with tabs and similar stuff the PCB contour is complex.
PcbDraw time needed to process these cases increases exponentially.
This patch solves a couple of issues related to extracting the PCB contour:

1. The algorithm used to find the closest element to a point.
2. Compute the contour just once (not twice)

### CODE REVIEW

File: pcbdraw/plot.py [hunk 1]
```diff
@@ -135,15 +136,19 @@ def flip(self) -> None:
 def matrix(data: List[List[Numeric]]) -> Matrix:
     return np.array(data, dtype=np.float32)
 
-def pseudo_distance(a: Point, b: Point) -> Numeric:
-    return (a[0] - b[0])**2 + (a[1] - b[1])**2
+# def distance(a: Point, b: Point) -> Numeric:
+#     return math.sqrt((a[0] - b[0])**2 + (a[1] - b[1])**2)
```
Comment (yaqwsx) [inline pcbdraw/plot.py:140]: What's the point of commenting out this function? At the moment, it is not used. However, it makes sense to have it ready to complement `pseudo_distance`. And if we want to get rid of it, we should delete it.

**Comment by yaqwsx** [comment]
Hi! Thanks for the optimization. I have a suggestion not to use commented out code.

Nevertheless, the true cost is in the following routine: https://github.com/yaqwsx/PcbDraw/blob/d2af9d44c998a3ba49c1880aa27bf4479697e5f9/pcbdraw/plot.py#L429-L469

It is cubic to the number of input segments. It could be done in n log(n). However, that would require the use of some space segmentation trees. I wanted to preserve the simplicity of the code, and I was counting on the outline being quite simple. This is, however, not the case with panels without reconstructed arcs.

File: pcbdraw/plot.py (hunk 1 above)
Comment (set-soft) [reply pcbdraw/plot.py:140]: I commented it because it isn't used, you can uncomment it when needed. Or just as a reference of what a real distance is.

I usually uncomment unused functions to avoid confusion (people might think this is used), misleading coverage results (code not covered just because is dead), avoid wasting resources, etc.

**Comment by set-soft** [comment]
> Nevertheless, the true cost is in the following routine:

Yes, this is exactly why this patch optimizes *get_closest*, the bottle neck of the code you mention.

**Comment by yaqwsx** [comment]
This should be addressed by afb3b8ba18efe036a53d40f77a70fff69658f5c9. The commit still needs further testing to ensure there are no edge cases breaking it.
